pkg/process/metadata/parser/java/spring.go

```diff
@@ -0,0 +1,205 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package javaparser contains functions to autodetect service name for java applications
+package javaparser
+
+import (
+	"archive/zip"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rickar/props"
+	"github.com/vibrantbyte/go-antpath/antpath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	defaultLocations       = "optional:classpath:/;optional:classpath:/config/;optional:file:./;optional:file:./config/;optional:file:./config/*/"
+	defaultConfigName      = "application"
+	locationPropName       = "spring.config.locations"
+	configPropName         = "spring.config.name"
+	activeProfilesPropName = "spring.profiles.active"
+
+	appnamePropName = "spring.application.name"
+)
+
+// parseURI parses locations (usually specified by the property locationPropName) given the list of active profiles (specified by activeProfilesPropName)
+// and the current directory cwd.
+// It returns a couple of maps each having as key the profile name ("" stands for default one) and as value the ant patterns where the properties should be found
+// The first map returned is the locations to be found in fs while the second map contains locations on the classpath (usually inside the application jar)
+func parseURI(locations []string, name string, profiles []string, cwd string) (map[string][]string, map[string][]string) {
```

Comment:
what's the performance of the method?
I do suspect the many allocations here will damage the performance

Reply:
it runs once per spring boot application and usually it has to transform this kind of string `optional:classpath:/;optional:classpath:/config/;optional:file:./;optional:file:./config/;optional:file:./config/*/"`

I think that can be acceptable to keep as it even not super optimized?